File: Easy/0258_add-digits/add-digits.cpp
class Solution
{
public:
	int addDigits(int num)
	{
		int n = num;
		while (n > 9)
		{
			int sum = 0;
			while (n)
			{
				sum += n % 10;
			  n /= 10;
			}
			n = sum;
		}
		return n;
	}
};

/*int addDigits(int num)*/
/*{*/
/*	if (num == 0)*/
/*		return 0;*/
/*	return 1 + (num - 1) % 9;*/
/*}*/

File: Easy/0258_add-digits/add-digits.py
#!/bin/python3

from typing import List

class Solution:
    def addDigits(self, num: int) -> int:
        if num == 0:
            return 0
        return 1 + (num - 1) % 9

def main():
    tests: List[int] = [
        38,
        0,
    ]

    for n in tests:
        print(f"---\nnum: {n}\naddDigits: {Solution().addDigits(n)}")

if __name__ == "__main__":
    main()
